tensorflow_decision_forests/contrib/temporal_feature_processor/temporal_feature_processor/implementation/pandas/operators/window/simple_moving_average.py

```diff
@@ -14,32 +14,72 @@
 
 """Implementation for the SimpleMovingAverage operator."""
 
+from typing import Dict
+
+import pandas as pd
+
 from temporal_feature_processor.implementation.pandas.data.event import PandasEvent
 from temporal_feature_processor.implementation.pandas.data.sampling import PandasSampling
 from temporal_feature_processor.implementation.pandas.operators.window.base import PandasWindowOperator
 
 
 class PandasSimpleMovingAverageOperator(PandasWindowOperator):
-  """Base class for window operators."""
+  """Pandas implementation for the simple moving average operator."""
 
-  def __init__(self, window_length: int) -> None:
+  def __init__(self, window_length: str) -> None:
     super().__init__(window_length=window_length)
 
   def __call__(
       self,
       data: PandasEvent,
       sampling: PandasSampling,
-  ) -> PandasEvent:
+  ) -> Dict[str, PandasEvent]:
     """Apply a simple moving average to an event.
 
     If input has more than one feature, the moving average will be computed for
     each of its features independently.
 
     Args:
-        data (PandasEvent): the input event to apply a simple moving average to.
+      data: the input event to apply a simple moving average to.
+      sampling: the desired sampling for the output event.
 
     Returns:
-        PandasEvent: the output of the operator.
+      Dict[str, PandasEvent]: the output event of the operator.
     """
+    # remove index to be able to filter using index values
+    data_no_index = data.reset_index()
+
+    # get index columns and name of timestamp column
+    index_columns = sampling.names[:-1]
```

Comment:
Can you document somewhere that the timestamps is always the last index (or alternatively, what about creating a utility function doing this extraction)?

Reply:
@getchebarne actually implemented that utility function [here](https://github.com/tensorflow/decision-forests/blob/main/tensorflow_decision_forests/contrib/temporal_feature_processor/temporal_feature_processor/implementation/pandas/operators/base.py#L34) but I missed it 😅 will make the change to use it.

We could additionally add a check to ensure that column is in fact a datetime one.